tracing-appender/src/non_blocking.rs

```diff
@@ -8,14 +8,52 @@ use std::sync::Arc;
 use std::thread::JoinHandle;
 use tracing_subscriber::fmt::MakeWriter;
 
+/// The default maximum number of buffered log lines.
+///
+/// If [NonBlocking][non-blocking] is configured to be lossy, it will drop any additional logs emitted when at
+/// capacity. If it is not lossy, backpressure will be exerted on senders, causing them to wait
+/// until there is buffer capacity remaining before enqueuing new lines.
+///
+/// [non-blocking]: ./struct.NonBlocking.html
+/// Recommended to be a power of 2.
 pub const DEFAULT_BUFFERED_LINES_LIMIT: usize = 128_000;
 
+/// A guard which triggers an associated [`NonBlocking`][non-blocking] writer to flush logs when dropped.
+///
+/// Writing to a [`NonBlocking`][non-blocking] writer will **not** immediately write a log line to the underlying
+/// output. Instead, the log line will be enqueued to be written by the logging worker thread. In
+/// addition, to improve throughput, the non-blocking writer flushes the underlying output
+/// periodically, rather than every time a line is written. This means that if the program
+/// terminates abruptly (such as by panicking, or by calling `std::process::exit`), some log lines
+/// may not be written.
+///
+/// [non-blocking]: ./struct.NonBlocking.html
```

Comment:
Nit, take it or leave it: the string ````NonBlocking```` is only one character longer than the string `non-blocking`; I don't think renaming the link reference here is really necessary
```suggestion
/// A guard which triggers an associated [`NonBlocking`] writer to flush logs when dropped.
///
/// Writing to a [`NonBlocking`] writer will **not** immediately write a log line to the underlying
/// output. Instead, the log line will be enqueued to be written by the logging worker thread. In
/// addition, to improve throughput, the non-blocking writer flushes the underlying output
/// periodically, rather than every time a line is written. This means that if the program
/// terminates abruptly (such as by panicking, or by calling `std::process::exit`), some log lines
/// may not be written.
///
/// [`NonBlocking`]: ./struct.NonBlocking.html
```
but, not a big deal.